pkg/controller/master/node/controller.go

```diff
@@ -0,0 +1,113 @@
+package node
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/harvester/pkg/config"
+	appsv1 "github.com/rancher/wrangler-api/pkg/generated/controllers/apps/v1"
+	v1 "github.com/rancher/wrangler-api/pkg/generated/controllers/core/v1"
+	apiv1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	controllerAgentName = "balance-minio-node-controller"
```

Comment:
probably is better to rename the file name to be more specific like `minio_controller.go` since this is `balance-minio-node-controller`